src/compiler/checker.ts

```
@@ -47937,7 +47937,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             case SyntaxKind.MethodSignature:
                 return checkMethodDeclaration(node as MethodDeclaration | MethodSignature);
             case SyntaxKind.ClassStaticBlockDeclaration:
-                return checkClassStaticBlockDeclaration(node as ClassStaticBlockDeclaration);
+                return checkNodeDeferred(node);

```

Maybe a stupid question, but why is it that this matters here, yet MethodSignatures are not deferred?

---

As in, can't we write that same local type declaration in a method and it breaks? Or is that fine because the _body_ gets deferred?

---

My understanding is that:
1. The same local type declaration in a method shouldn't suppress the error either.
2. Object literal methods are deferred but class methods are not. If I try to defer class methods... a ton of tests break. So there has to be a reason behind why they are not deferred. Perhaps it's important for the variance worker?
3. Class static blocks conceptually execute after the class declaration and they shouldn't affect the variance measurement and shouldn't trigger circularities in a way class methods/properties can. So it should be OK to defer them. They should behave in an almost same way as statements executed after the class. They only get a special treatment when it comes to class type parameters and `this`. But it feels like it shouldn't break anything if those get simply resolved in a deferred manner - after the class declaration gets processed.
4. Right now it looks like this PR doesn't do anything because a bug has been introduced in the meantime: https://github.com/microsoft/TypeScript/issues/61982